registry/remote/auth/example_test.go

```diff
@@ -0,0 +1,248 @@
+	clientConfigTargetURL string
+	tokenScopes           = []string{
+		"repository:dst:pull,push",
+		"repository:src:pull",
+	}
+)
+
+func TestMain(m *testing.M) {
+	// create an authorization server
+	as := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.WriteHeader(http.StatusUnauthorized)
+			panic("unexecuted attempt of authorization service")
+		}
+		header := "Basic " + base64.StdEncoding.EncodeToString([]byte(username+":"+password))
+		if auth := r.Header.Get("Authorization"); auth != header {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		if got := r.URL.Query().Get("service"); got != host {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		if got := r.URL.Query()["scope"]; !reflect.DeepEqual(got, tokenScopes) {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		if _, err := fmt.Fprintf(w, `{"access_token":%q}`, accessToken); err != nil {
+			panic(err)
+		}
+	}))
+	defer as.Close()
+
+	// create a test server
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		path := r.URL.Path
+		if r.Method != http.MethodGet {
+			w.WriteHeader(http.StatusNotFound)
+			panic("unexpected access")
+		}
+		switch path {
+		case "/basicAuth":
+			wantedAuthHeader := "Basic " + base64.StdEncoding.EncodeToString([]byte(username+":"+password))
+			authHeader := r.Header.Get("Authorization")
+			if authHeader != wantedAuthHeader {
+				w.Header().Set("Www-Authenticate", `Basic realm="Test Server"`)
+				w.WriteHeader(http.StatusUnauthorized)
+			}
+		case "/accessToken":
+			wantedAuthHeader := "Bearer " + accessToken
+			if auth := r.Header.Get("Authorization"); auth != wantedAuthHeader {
+				challenge := fmt.Sprintf("Bearer realm=%q,service=%q,scope=%q", as.URL, host, strings.Join(tokenScopes, " "))
+				w.Header().Set("Www-Authenticate", challenge)
+				w.WriteHeader(http.StatusUnauthorized)
+			}
+		case "/clientConfig":
+			wantedAuthHeader := "Basic " + base64.StdEncoding.EncodeToString([]byte(username+":"+password))
+			authHeader := r.Header.Get("Authorization")
+			if authHeader != wantedAuthHeader {
+				w.Header().Set("Www-Authenticate", `Basic realm="Test Server"`)
+				w.WriteHeader(http.StatusUnauthorized)
+			}
+		case "/simple":
+			w.WriteHeader(http.StatusOK)
+		default:
+			w.WriteHeader(http.StatusNotAcceptable)
+		}
+	}))
+	defer ts.Close()
+	host = ts.URL
+	uri, _ := url.Parse(host)
+	expectedHostAddress = uri.Host
+	targetURL = fmt.Sprintf("%s/simple", host)
+	basicAuthTargetURL = fmt.Sprintf("%s/basicAuth", host)
+	accessTokenTargetURL = fmt.Sprintf("%s/accessToken", host)
+	clientConfigTargetURL = fmt.Sprintf("%s/clientConfig", host)
+	http.DefaultClient = ts.Client()
+
+	os.Exit(m.Run())
+}
+
+// ExampleClient_Do_minimalClient gives an example of a minimal working client.
+func ExampleClient_Do_minimalClient() {
+	var client auth.Client
+	// targetURL is of form http://ipaddr:port
```

Comment:
`targetURL` can be any URL. For example, `https://registry.wabbit-networks.io/v2/`

Reply:
Changed accordingly.